test/throttle_SUITE.erl

```diff
@@ -0,0 +1,214 @@
+-module(throttle_SUITE).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-compile([export_all, nowarn_export_all]).
+
+-define(DEFAULT_NO_PROCESSES, 10).
+-define(DEFAULT_INTERVAL, 60000). %% one minute
+
+all() ->
+    [
+     {group, api}
+    ].
+
+groups() ->
+    [
+     {api, [parallel],
+      [
+       start,
+       rate_zero_is_not_accepted,
+       low_rate_gets_remapped,
+       low_interval_get_remapped,
+       start_and_stop,
+       change_rate,
+       change_rate_gradually,
+       send_and_wait,
+       run_with_interval_zero_limits_only_number_of_parallel_executions,
+       pause_and_resume,
+       get_state
+     ]}
+    ].
+
+init_per_suite(Config) ->
+    application:ensure_all_started(amoc),
+    amoc_cluster:set_master_node(node()),
+    TelemetryEvents = [[amoc, throttle, Event] || Event <- [init, rate, request, execute, process]],
+    telemetry_helpers:start(TelemetryEvents),
+    Config.
+
+end_per_suite(_) ->
+    application:stop(amoc),
+    telemetry_helpers:stop(),
```

Comment:
currently `telemetry_helpers:stop()` doesn't unload meck module, also we want to reset meck module on init_per_testcase to ensure that test cases do not affect each other.

Reply:
We shouldn't need to. Test cases are running in parallel here, and also, when checking history we're filtering for the testcase name.